### PR TITLE
adds flow graph visualization to app workflows

### PR DESCRIPTION
**Pull request by astewes**

### CODE REVIEW

File: cdap-ui/app/services/helpers.js [hunk 1]
```diff
@@ -97,12 +97,62 @@ angular.module(PKG.name+'.services')
...
+    for(var i=0; i<nodes.length; i++) {
+      if (nodes[i].nodeType === 'ACTION') {
+        expandedNodes.push(nodes[i]);
+      } else if (nodes[i].nodeType === 'FORK') {
+        for (var j=0; j<nodes[i].branches.length; j++) {
+          expandForks(nodes[i].branches[j], expandedNodes);
+        }
+      }
+    }
+  }
```
Comment (gaarf) [inline cdap-ui/app/services/helpers.js]: these 3 functions are very specific to the API being used... IMHO they should not live in `myHelpers`. For now I would put them in the controller file that uses them, or if they are going to be re-used, in a different service. Later this belongs in a `myWorkflowsApi`.

File: cdap-ui/app/services/helpers.js [hunk 2]
```diff
@@ -97,12 +97,62 @@ angular.module(PKG.name+'.services')
...
+
+  function flatten(source, fork, target, connections) {
+    // Flatten a source-fork-target combo to [list of connections]
+    var branches = fork.branches,
+        temp = [];
+
+    for (var i =0; i<branches.length; i++) {
+      temp = branches[i];
+      temp.unshift(source);
+      temp.push(target);
```
Comment (gaarf) [inline cdap-ui/app/services/helpers.js]: use `Array.splice` to do both operations in one statement.

File: cdap-ui/app/features/workflows/controllers/tabs/runs/flow-ctrl.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 angular.module(PKG.name + '.feature.workflows')
```
Comment (nistala) [inline cdap-ui/app/features/workflows/controllers/tabs/runs/flow-ctrl.js]: should call this file workflow-ctrl

File: cdap-ui/app/services/helpers.js (hunk 2 above)
Comment (ajainarayanan) [inline cdap-ui/app/services/helpers.js]: As in like this? ```a.splice(0, 0, 0).splice(a.length-1, 0 , 2)```

Comment (gaarf) [inline cdap-ui/app/services/helpers.js]: you're right, splice is useless here. how about:
`temp = [source].concat(branches[i], [target]);`

File: cdap-ui/app/features/workflows/controllers/tabs/runs/status-ctrl.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
+angular.module(PKG.name + '.feature.workflows')
+  .controller('WorkflowsDetailRunStatusController', function($state, $scope, MyDataSource, amMoment, $filter) {
+    var dataSrc = new MyDataSource($scope),
+        filterFilter = $filter('filter'),
+        basePath = '/apps/' + $state.params.appId + '/workflows/' + $state.params.programId;
+    $scope.moment = amMoment;
+    $scope.moment.changeLocale('en');
```
Comment (gaarf) [inline cdap-ui/app/features/workflows/controllers/tabs/runs/status-ctrl.js]: I'm assuming that would change the locale for all amMoment instances throughout the app. What is it for, why is it here?

**Comment by gaarf** [comment]
is `$scope.moment` used anywhere ? Also the utility functions can go outside of the controller's closure, I can show you how.

**Comment by gaarf** [comment]
LGTM :+1:

**Pull request merged**